src/software/geom/algorithms/step_along_perimeter.h

```diff
@@ -8,13 +8,13 @@
  * Computes the final position on a polygon's perimeter after traveling a specified
  * distance from a starting point.
  *
- * Given a polygon, a starting point on its perimeter, and a travel distance,
- * this function calculates the final point reached upon traveling the distance along the
- * perimeter. Travel direction is determined by the sign of the distance: positive travels
- * clockwise, while negative travels counterclockwise.
+ * Given a polygon, a starting point, and a travel distance, this function calculates the
+ * final point reached upon traveling the distance along the perimeter starting from the
+ * point on the perimeter closest to the start point. Travel direction is determined by
+ * the sign of the distance: positive travels clockwise, while negative travels
+ * counterclockwise.
  *
- * @note Assumes that the starting point is exactly on the polygon's perimeter. If
- * distance is zero, the function returns the starting point.
+ * @note If the distance is zero, the function returns the starting point.
```

Comment:
Given the function name, I think it would make more sense to return the closest point on the perimeter if the given distance is 0